### PR TITLE
Added sink with emitter, so that it can be used to index, update and …

### PR DESCRIPTION
**Pull request by arunx2**
Need some refactoring and cleanup on the emitter. It is more aligned with elasticsearch sink connector to allow index, update and delete documents from/to the index. I'll add more safety mechanisms and test cases later. 

With this we can also extend this to flink's table APIs

### CODE REVIEW

File: flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/OpenSearchSinkWriter.java [hunk 1]
```diff
@@ -121,6 +122,11 @@ protected void submitRequestEntries(
                     "Bulk index of {} documents took {}",
                     requestEntries.size(),
                     bulkResponse.getIngestTookInMillis());
+            List<String> failedEntries = new ArrayList<>();
+            //            for(BulkItemResponse bir: bulkResponse.getItems()){
+            //               if (needReprocess(bir)) {
+            //                   failedEntries.add(bir.)
+            //            }
```
Comment (jeremyber-aws) [inline flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/OpenSearchSinkWriter.java]: not implemented yet?

File: flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/sink/OpenSearchWriter.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        // configured or shutdown without a final
+        // checkpoint) the mailbox might already be shutdown, so we should not enqueue any
+        // actions.
+        if (isClosed()) {
+            return;
+        }
+        mailboxExecutor.execute(action, actionName);
+    }
+
+    private void extractFailures(BulkRequest request, BulkResponse response) {
```
Comment (jeremyber-aws) [inline flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/sink/OpenSearchWriter.java:284]: does this do anything to know which failures are "retryable"?

File: flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/OpenSearchSinkWriter.java (hunk 1 above)
Comment (arunx2) [reply flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/OpenSearchSinkWriter.java]: I removed it. New SinkWriter is in sink package!

File: flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/sink/OpenSearchWriter.java (hunk 2 above)
Comment (arunx2) [reply flink-connectors/flink-connector-opensearch/src/main/java/org/apache/flink/streaming/connectors/opensearch/sink/OpenSearchWriter.java:284]: As of now, It doesn't classify the failures. There will be a retry for all failures. But it honors the bulkBackoffMaxRetries which is set to 3 by default.

**Comment by arunx2** [comment]
I'll add more samples for update and delete requests. I think it will be complete when we add support to dynamic sink to support SQL.

**Pull request merged**